File: app/src/main/java/net/myacxy/squinch/models/DebugLogEntry.java
package net.myacxy.squinch.models;

import android.databinding.BaseObservable;
import android.databinding.Bindable;
import android.support.annotation.StringDef;

public class DebugLogEntry extends BaseObservable {

    public static final String TYPE_LOG = "Log";
    public static final String TYPE_EXCEPTION = "Exception";
    public static final String TYPE_USER = "User";

    private long time;
    private String type;
    private String message;

    public DebugLogEntry(long time, String type, String message) {
        this.time = time;
        this.type = type;
        this.message = message;
    }

    @Bindable
    public long getTime() {
        return time;
    }

    @Bindable
    public String getType() {
        return type;
    }

    @Bindable
    public String getMessage() {
        return message;
    }

    @StringDef({TYPE_LOG, TYPE_EXCEPTION, TYPE_USER})
    public @interface DebugLogEntryType {
    }
}

File: app/src/main/java/net/myacxy/squinch/settings/debuglog/DebugLogEntry.kt
package net.myacxy.squinch.settings.debuglog

import android.support.annotation.StringDef

class DebugLogEntry(val time: Long, @DebugLogEntryType val type: String, val message: String) : Comparable<DebugLogEntry> {

    override fun compareTo(other: DebugLogEntry): Int {
        return time.compareTo(other.time)
    }

    @StringDef(TYPE_LOG, TYPE_EXCEPTION, TYPE_USER)
    annotation class DebugLogEntryType

    companion object {

        const val TYPE_LOG = "Log"
        const val TYPE_EXCEPTION = "Exception"
        const val TYPE_USER = "User"
    }
}
